editors/code/src/config.ts

```diff
@@ -46,7 +47,9 @@ export class Config {
 
     private refreshConfig() {
         this.cfg = vscode.workspace.getConfiguration(Config.rootSection);
-        console.log("Using configuration:", this.cfg);
+        const enableLogging = this.cfg.get("trace.extension") as boolean;
+        log.setEnabled(enableLogging);
```

Comment:
I am not sure we should put it here... We use `activate*()`  where we setup related event listeners, so why won't we do that for logging?

Reply:
Logging is an odd thing, because it needs to be configured early, earlier than everything else. Hence, explicitelly pushing config from config replace seems a better solution than trying to do it elsewhere, implicitelly hopeing that it'll happen early enough

Reply:
Hmm, I think there is a bug here. The order of `onDidChangeConfiguration()` event listeners execution is undefined (I suppose), other parts of the code use it and may not get the fresh config (i.e. `refreshConfig()` is not executed earlier then other listeners)...